### PR TITLE
Fr us 20

### LINKED ISSUE #23: User Story 20, User Can Edit their Profile Data
As a registered user
When I visit my profile page
I see a link to edit my profile data
When I click on the link to edit my profile data
I see a form like the registration page
The form is prepopulated with all my current information except my password
When I change any or all of that information
And I submit the form
Then I am returned to my profile page
And I see a flash message telling me that my data is updated
And I see my updated information

### PR DESCRIPTION
**Pull request by fredrondina**
completed US 20. User detail can now be updated from /profile. Password update will be added in us 21, email update will be added in us 22

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -16,11 +16,27 @@ def create
...
+  end
+
   def show
     user = User.find(current_user.id)
     flash[:success] = "Logged in as #{user.user_detail.name}"
   end
 
+  def update
+    user = current_user
+    user.user_detail.update(user_params[:user_detail_attributes])
```
Comment (brian-greeson) [inline app/controllers/users_controller.rb:30]: This is fine for now

**Review by brian-greeson — APPROVED** [review]
This looks good. Glad we figured out the update issue with validating password

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -16,11 +16,27 @@ def create
...
+    if user.save
+      flash[:notice] = "Your information has been updated."
+      redirect_to '/profile'
+    else
+      flash[:error] = user_detail.errors.full_messages.to_sentence
```
Comment (kelshadarby) [inline app/controllers/users_controller.rb:35]: I think we can clean this up in a later refactor :D

File: app/models/user.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ class User < ApplicationRecord
 
   validates_presence_of :email_address
   validates :email_address, uniqueness: {message: "Email is already in use."}
-  validates_presence_of :password, require: true
```
Comment (kelshadarby) [inline app/models/user.rb:6]: Nice catch here!

File: app/views/users/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  <%= text_field_tag "user_detail_attributes[street_address]", "#{@user.user_detail.street_address}" %>
+  <%= label_tag :city %>
+  <%= text_field_tag "user_detail_attributes[city]", "#{@user.user_detail.city}" %>
+  <%= label_tag :state %>
+  <%= text_field_tag "user_detail_attributes[state]", "#{@user.user_detail.state}" %>
+  <%= label_tag :zip_code, "Zip Code" %>
+  <%= text_field_tag "user_detail_attributes[zip_code]", "#{@user.user_detail.zip_code}" %>
+  <%= label_tag :email_address, "Email Address" %>
+  <%= text_field_tag :email_address, "#{@user.email_address}" %>
+<%= submit_tag "Update Info" %>
```
Comment (kelshadarby) [inline app/views/users/edit.html.erb:15]: Looks very clean!

**Review by kelshadarby — APPROVED** [review]

**Pull request merged**